db/database.py

```diff
@@ -0,0 +1,192 @@
+        i = rooms.insert()
+        for room in self.my_amity.rooms:
+            if room not in self.already_added:
+                if room.room_type == "Office":
+                    is_office = True
+                else:
+                    is_office = False
+                if room in self.my_amity.vacant_rooms:
+                    is_vacant = True
+                else:
+                    is_vacant = False
+                i.execute(
+                    name=room.name,
+                    is_office=is_office,
+                    is_vacant=is_vacant
+                )
+                self.already_added.append(room)
+
+    def add_allocations(self):
+        """
+        Add data from the Allocations list to the database.
+        Each room gets a database table with occupants as rows.
+        """
+        if self.my_amity.rooms:
+            for room in self.my_amity.rooms:
+                room_name = room.name
+                room_name = Table(
+                        room_name, self.metadata,
+                        Column(
+                            "employee_id", Integer,
+                            ForeignKey("people.employee_id"),
+                            nullable=False),
+                        Column("name", String),
+                        extend_existing=True
+                )
+                if room_name.exists():
+                    room_name.update()
+                else:
+                    room_name.create()
+                i = room_name.insert()
+                if room.occupants:
+                    for person in room.occupants:
+                        i.execute(
+                            employee_id=person.emp_id,
+                            name=person.name
+                        )
+
+    def load_state(self, args):  # pragma: no cover
+        """Load data to the application from a user-defined database"""
+        self.db_name = "sqlite:///" + "%s" % args.get("<sqlite_database>")
+        self.db = create_engine(self.db_name)
+        connection = self.db.connect()
+        people_ = connection.execute("SELECT * FROM people")
+        rooms_ = connection.execute("SELECT * FROM rooms")
+        table_names = connection.execute(
+            "SELECT name FROM sqlite_master \
+            WHERE (type='table') AND (name NOT LIKE 'people') \
+            AND (name NOT LIKE 'rooms') "
+        )
+
+        for row in people_:
+            print('people in load_state', people_)
+            if row["is_fellow"]:
+                is_fellow = True
+            else:
+                is_fellow = False
+            first_name = row["first_name"]
+            last_name = row["last_name"]
+            emp_id = row["employee_id"]
+
+            self.my_amity.add_person_from_db({
+                    "is_fellow": is_fellow,
+                    "first_name": first_name,
+                    "last_name": last_name,
+                    "is_fellow": is_fellow,
+                    "emp_id": emp_id
```

Comment:
The `emp_id` variable is used to populate the `emp_id` field of the Person class. It's used as the person identifier when reallocating rooms.